test/json_unittest.cc

```diff
@@ -0,0 +1,385 @@
+#include <functional>
+
+#include "../src/json.h"
+#include "gtest/gtest.h"
+
+#define EXPECT_TO_STRING(v, s) EXPECT_EQ(v->ToString(), s)
+
+namespace {
+
+void GuardJsonException(std::function<void()> test) {
+  try {
+    test();
+  } catch (const json::Exception& exc) {
+    FAIL() << exc.what();
+  }
+}
+
+
+// Trival Null
```

Comment:
Nit: missing trailing ".". Also everywhere below.

Reply:
They are titles, not sentences :)

Reply:
https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar